---
title: C28105
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
  - "C28105"
helpviewer_keywords:
  - "C28105"
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
  - "multiple"
---
# C28105
Warning C28105: Leaking resource due to an exception

 The specified resource is not freed when an exception is raised. The statement specified by the path can raise an exception. This warning is similar to warning [C28103](../code-quality/c28103.md), except that in this case an exception is involved.

## Example
 The following code example generates this warning:

```
res = KeSaveFloatingPointState(buffer);

res = AllocateResource(Resource);
char *p2 = new char[10]; // could throw

delete[] p2;
FreeResource(Resource)
```

 The following code example avoids this warning:

```
res = AllocateResource(Resource);
char *p2;

try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
FreeResource(Resource)
```